### ISSUE TITLE
Homey Pro (early2023) can't connect to BLE curtains

### ISSUE THREAD

**Issue by moezie** [AdyRock/com.switchbot#59]
Hi there,

The Homey won't add one of my BLE curtains, it keeps telling me there are no devices found. One of my other BLE curtains that are added in the Homey from the first migration, isn't working properly. Sometimes it works but most of the time the commands send from the Homey are not received by the BLE curtains. Even when they are in near proximity (less then 2 meters). 

I've also noticed that the position of the curtains is not updated since the migration to the new Homey Pro.

I'm currently running Homey version 10.0.0-rc.62 with the test version of the Switchbot app (v1.1.24).

**Comment by DrJaws2** [comment]
Hi there; I have the same issue; on the early 2023 Pro running 10.0.rc.78. Thanks for your help!

**Comment by AdyRock** [comment]
Sorry for taking so long to reply. 
Could you both send the Log from the App Settings page and let me know when you have sent them.

**Comment by moezie** [comment]
No problem, thanks for reaching out! :)

I've just send the log from the app settings page. The reference I got is 29ea1789-6fe4-4a8b-916f-ae88ca158778.

**Comment by DrJaws2** [comment]
I also have just sent the log file (seconds ago)
Thx for reaching out!

**Comment by AdyRock** [comment]
@moezie I haven't got a log from you and the code doesn't look familiar. 

@DrJaws2 The errors showing in your log relate to the cloud with OAuth issues. Have you tried repairing one of the devices to see if that helps.

**Comment by DrJaws2** [comment]
Yes I have tried everything as far as I can: resetting to factory settings; pairing mode; Homey just doesn’t seem to pick up the BLE signal

**Comment by DrJaws2** [comment]
The log indeed shows errors when trying to connect via the other option (hub) yet since I don’t have a hub this error occurs.

**Comment by AdyRock** [comment]
Could you set the Log option to Detailed information, try adding a BLE device, and then send the log again. Hopefully it will catch some BLE details.

**Comment by moezie** [comment]
@AdyRock oh my bad, I used the log option of Homey. I have just sent the log using the App Settings screen.

**Comment by AdyRock** [comment]
@moezie So, looking at your log, it seems you have the same issue that I am now seeing with "Error: Software caused connection abort" when you try to control a device. I have reported this on Slack and hope someone from Athom will look into it.
I can also see the "serviceData" is not being returned in one of the BLE responses and that is where the information is that identifies the type of device and status. That is normally caused by a weak signal. Unfortunately, the HP2023 has a bug so the RSSI data is not being provided, so I can't see what the signal strength is like.

Do you have a Blind / Tilt motor on BLE?

**Comment by AdyRock** [comment]
I have published a new test version that supports Blind / Tilt firmware v2. This removes the requirement of getting the position by sending a command as it is now supplied in the advertisement data.

**Comment by moezie** [comment]
Lets hope so! I also tried to add the curtain motor while holding the motor next to the Homey. I can try that again to see if the log shows other data? I don't have the blind/tilt motor unfortunately.

**Comment by AdyRock** [comment]
I was just wondering if the Blind / Tilt was causing a problem because it had to be polled by sending a command rather than passively listening to the broadcasts.

**Comment by AdyRock** [comment]
One thing you could do is find the BLE mac address from the Switchbot app, then go to the developer page (https://tools.developer.homey.app/tools/ble) and try to find it.

**Comment by DrJaws2** [comment]
I can find the device via the MAC address; but unable to connect to it.

**Comment by DrJaws2** [comment]
I've sent a new log

**Comment by AdyRock** [comment]
Connections are only temporary as it will kick you out after a few seconds. Luckily a connection is only required to send commands. All the status data is in the Advertised Service data, which is broadcast frequently.

**Comment by AdyRock** [comment]
I haven't received a log. Did you send it from the app settings page or was it a Diagnostics report?

**Comment by moezie** [comment]
I can find the MAC addresses of the two curtain BLE in the developer page but I am not able to connect to the missing one.  I am getting the message "Error running command CONNECT: Software caused connection abort" when trying to connect to the missing one. Connecting to the linked curtain BLE doesn't seem to give me any response.

**Comment by DrJaws2** [comment]
from the app settings page; was send successfully (full information)
have repeated it another time; again successfully

**Comment by DrJaws2** [comment]
main thing for me is now the grouping/ungrouping of the curtains: seems to make it easier to keep them grouped and then command the dominant curtain in the flow. And the grouping (and possible other mare advanced settings) seem to be exclusively editable from the Switchbot-app (iOS). But that should be workable as well.

**Comment by AdyRock** [comment]
Just out of interest, when you found the device on the developer page, could you see the Advertised Service data?
If the app can't see that then it doesn't recognise it as that is where the identification codes are.

**Comment by moezie** [comment]
No, it shows an empty array as the advertised service data for both BLE curtains :(

**Comment by DrJaws2** [comment]
yes: Advertised service data (hex)
[{"uuid":"0000fd3d-0000-1000-8000-00805f9b34fb","data":"63c046642205"}]

**Comment by moezie** [comment]
@DrJaws2 thats nice! Did you do a reset of the curtains or something in that order?

**Comment by DrJaws2** [comment]
no..... I thought you changed something..... haha

**Comment by DrJaws2** [comment]
I don't know why; but after finding the curtains in de developer BLE page they somehow showed up in the app itself. Without changing major things at the curtains themselves. At least not to my knowledge.

**Comment by AdyRock** [comment]
Missing service data in the past is normally a sign of poor signal, but I obviously don't have a lot of experience with HP2023. As I'm also seeing other BLE issues with HP2023 it could be all related. 

One other question is are you using the Web app or the phone app?
I have also seen the web app timeout if it doesn't get a reply within 10 seconds but the phone app can go for 30 seconds.

**Comment by DrJaws2** [comment]
in generall the phone app

**Comment by DrJaws2** [comment]
I mostly now see a non-logic relation between the controls and the curtain itself. I've two sets of curtains and they also seem te react differently within the app; but I'm not certain now where the issue seems to come from: I'll fully charge them first and then will try so see if they will react logically after giving them some time

**Comment by DrJaws2** [comment]
Within the iOS SwitchBot app the current position of the curtain is displayed accurately contrary to the information displayed in the Homey app -> devices -> specific curtain 'hard-press' -> position
Battery information seems accurate.
Within the app itself two curtains remain bold displayed; two remain faded: no clue why and/or how to change it (the on/off switch doesn't seem to do the job).

**Comment by moezie** [comment]
I just had a brainwave. My wifi network for domotic stuff is only using 2.4Ghz which is the one used by the HP2023 and it could interfere with bluetooth. I added the 5Ghz band to the domotic wifi and reconnected the HP2023. Now I am able to connect to the BLE curtains using the developer BLE page. So it seems like the wifi connection was interfering with the bluetooth connections. However the app still won't find the missing BLE curtain :(.

**Comment by AdyRock** [comment]
The main thing is if you see the Service data.

**Comment by DrJaws2** [comment]
It seems I'm now able to incorporate them in flows!

**Comment by moezie** [comment]
The already connected switchbot is showing service data now. The other still isn't. Also sending commands is still hit and miss so no changes there unfortunately.

**Comment by DrJaws2** [comment]
> It seems I'm now able to incorporate them in flows!

After a first flow; it seems the flow doesn't properly continues / repeats itself. I don't know if this has to do with the connection between the iOS app (switchbot) and the curtains | does that connection interfere with the flow commands from the homey-pro? Or that the trigger in my flow (lux from a Tempest weather flow station has issues....

**Comment by DrJaws2** [comment]
also the position information and battery information doesn't come through all te time.

**Comment by AdyRock** [comment]
The Switchbot app has no affect, unless both try to send a command at the same time. 
I think the issues you are seeing are due to Homey's problems with BLE.
Hopefully they will improve it at some point as they are working on it.

**Comment by DrJaws2** [comment]
regretfully its not workable: the machines got disconnected and it seems very difficult to get them connected again. If I've understood you correctly I'll need to wait for Homey to step in here? To bad: not able to get them in my flows; and the Switchbo flow options are very limited.......

**Comment by AdyRock** [comment]
Have you updated Homey to the latest version (105 I think it is)?

**Comment by AdyRock** [comment]
I would also highly recommend using a Switchbot mini hub as it has a good BLE connection to the devices and works very well. 
I know it relies on the Internet, which might be a problem if your connection is unreliable, but I have no issue with it, although I do use my ESP32 modules as backup.

**Comment by DrJaws2** [comment]
> Have you updated Homey to the latest version (105 I think it is)?

Was update this morning; haven’t been home yet: I’ll give it another try!

**Comment by DrJaws2** [comment]
> I would also highly recommend using a Switchbot mini hub as it has a good BLE connection to the devices and works very well.
> I know it relies on the Internet, which might be a problem if your connection is unreliable, but I have no issue with it, although I do use my ESP32 modules as backup.

The general thought underneath a homey off course is not needing extra hubs…… 🙃

**Comment by AdyRock** [comment]
Hmm, nice dream but not reality 😁
I have 9 other hubs that work alongside Homey (Somfy, LightWave, Hue, SmartThings, Switchbot x 3, Gardena and Linktap). I find they do a great job and Homey makes everything together nicely.

**Comment by AdyRock** [comment]
Just remembered I also have 2 Misol hubs.

**Comment by BerndGaykema** [comment]
I also have 2 sets of Curtain BLE (each set consists of 2 coupled curtain units) and have trouble in getting it to work.
I had it coupled to sun-setup and sun-down but 9 out of 10 it didn't work.
Now I have it coupled to a dark sensor but same issue. Extremely unreliable.

I have the Homey-Pro within one meter from one set and about 5 meters from the second set.

I often see a timeout of 300000msec (not sure about the number of zero's).

Is Homey-Pro (Early 2023) still not mature enough to get it rolling?

**Comment by BerndGaykema** [comment]
An update:
When I ungroup the 2 curtains and make each curtain a device in Homey, then it seems to work ok.
Looks like Homey can't handle grouped devices.

**Comment by AdyRock** [comment]
Hmm, seems strange as mine are grouped.
When you had them grouped were you trying to send commands to both?

**Comment by DrJaws2** [comment]
I have two pairs of BLE’s curtains: one pair is working perfectly now and the other refuses. Time out indeed every time. They are configured exactly the samen. The hub is positioned within five meters of all curtains….. it leaves me questioning why one pair works perfectly while the other refuses…..

**Comment by AdyRock** [comment]
When you say the hub, us that a Switchbot hub or Homey or an ESP32?

**Comment by BerndGaykema** [comment]
> Hmm, seems strange as mine are grouped. When you had them grouped were you trying to send commands to both?

No. Just one device (of the group) was addressed. I first tested which address (of both from the group) worked before I defined the device in Homey.

The only "problem" now is that when the second curtain closes and touched the already closed, this last one opens sometimes. But with the "low power" mode this seems solved as well.

**Comment by DrJaws2** [comment]
> When you say the hub, us that a Switchbot hub or Homey or an ESP32?

The Homey Pro
